pkg/chunkenc/memchunk.go

```diff
@@ -669,19 +682,24 @@ func (si *bufferedIterator) Error() error { return si.err }
 func (si *bufferedIterator) Close() error {
 	if !si.closed {
 		si.closed = true
-		si.pool.PutReader(si.reader)
-		BufReaderPool.Put(si.bufReader)
-		if si.buf != nil {
-			BytesBufferPool.Put(si.buf)
-		}
-		si.origBytes = nil
-		si.bufReader = nil
-		si.buf = nil
-		si.decBuf = nil
-		si.reader = nil
+		si.close()
 		return si.err
 	}
 	return si.err
```

Comment:
It seems like we don't need `return si.err` twice now

Reply:
true, I'm also going to improve so we can add more stats easily.